### ISSUE TITLE
Error with NumPy 2.1.3

### ISSUE THREAD

**Issue by jakobmoss** [BASTAcode/BASTA#56]
Running BASTA with new NumPy (tested with 2.1.3) results in an error:
```
ModuleNotFoundError: No module named 'numpy.lib.histograms
```

They changed the API last year including a renaming of the file we import (https://github.com/numpy/numpy/commit/89a6f3ee13c56a27b13cfe76f353f96f6695af28):

![Screenshot 2024-11-06 at 12 46 00](https://github.com/user-attachments/assets/8485fa4e-fc59-42a6-a222-5ab4dff69525)

This issue might be updated as I test BASTA with newest versions of our requirements (in prep for solving #54 ).

**Comment by mr-asher** [comment]
@jakobmoss, I ran into this issue too. I pulled BASTA, created a new virtual environment using Python 3.12 and found a few issues which I manually fixed and was able to run the code. 

The other one I remember was distutils no longer having a strbool function. My solution to this was to add a custom strbool function to the general utils file. 

Would it be useful if I made a pull request showing the changes I made?

**Comment by mr-asher** [comment]
https://github.com/BASTAcode/BASTA/pull/58 attempts to address this issue and a couple of others I've found. It's working on the example BASTArun given in the docs, but hasn't been tested beyond that.

**Comment by jakobmoss** [comment]
@mr-asher, excellent, thanks! I will have a look at it.

**Comment by jakobmoss** [comment]
**Note to self**

Another thing changed by NumPy version 2 is that as a default, the data type is included when printing. This makes sense from a typing and point of view and makes it very clear which data type a given variable is. However, this makes our log-file a bit cluttered and not as easy to read:

<img width="1290" alt="Screenshot 2024-11-10 at 11 41 21" src="https://github.com/user-attachments/assets/1ed5ac22-405b-407f-8c81-1fc3c3ef8a66">


It is easy to fix, as NumPy introduced the function `numpy.set_printoptions` in the release. Therefore, for now, I will add the following to the code (in `bastamain`):

```python
    # Enable legacy printing of NumPy data types
    # --> E.g., print 104.14836386995329 instead of np.float64(104.14836386995329)
    #     and 'Teff' instead of np.str_('Teff') to the .log file
    np.set_printoptions(legacy='1.25')
```

Which will bring the .log back to how it used to look:

<img width="742" alt="Screenshot 2024-11-10 at 11 41 38" src="https://github.com/user-attachments/assets/1322824d-44b1-454e-87f0-8e2792154417">

We might consider if we wish to change this in a future release.

**Comment by jakobmoss** [comment]
Included in the release [v1.5.0](https://github.com/BASTAcode/BASTA/releases/tag/v1.5.0) .
